client/app/main/school/school.reports.controller.js

```diff
@@ -252,6 +252,102 @@ function SchoolReportsCtrl($scope, $timeout, Auth, uiGridGroupingConstants, Stud
     });
   };
 
+  $scope.outreachesGridOptions = {
+    rowHeight: 27,
+    enableSorting: true,
+    enableGridMenu: true,
+    enableFiltering: true,
+    treeRowHeaderAlwaysVisible: false,
+    exporterPdfDefaultStyle: {fontSize: 9},
+    exporterPdfTableHeaderStyle: {fontSize: 10, bold: true, color: 'grey'},
+    exporterPdfHeader: {
+      text: 'Chronically Absent Students',
+      style: 'headerStyle'
+    },
+    exporterPdfOrientation: 'landscape',
+    exporterPdfPageSize: 'LETTER',
+    exporterPdfMaxGridWidth: 500,
+    exporterPdfFooter: function(currentPage, pageCount) {
+      return {
+        text: currentPage.toString() + ' of ' + pageCount.toString(),
+        style: 'footerStyle'
+      };
+    },
+    exporterPdfCustomFormatter: function(docDefinition) {
+      docDefinition.styles.headerStyle =
+      {fontSize: 22, bold: true, color: '#265E6D'};
+      docDefinition.styles.footerStyle = {fontSize: 10, bold: true};
+      return docDefinition;
+    }
+  };
+
+  $scope.outreachesGridOptions.columnDefs = [{
+    name: 'student.studentId',
+    displayName: 'Student Id',
+    minWidth: 150,
+    cellTemplate: '<div class="ui-grid-cell-contents">' +
+                  '<a href="/student/{{row.entity.student._id}}">' +
+                  '{{row.entity.student.studentId}}</a>' +
+                  '</div>'
+    }, {
+      name: 'student.firstName',
+      displayName: 'First Name',
+      minWidth: 150
+    }, {
+      name: 'student.lastName',
+      displayName: 'Last Name',
+      minWidth: 150
+  }, {
+      name: 'totals.all',
+      displayName: 'Total',
+      minWidth: 80
+  }, {
+      name: 'totals["Phone Call"]["count"] || 0',
+      displayName: 'Calls',
+      minWidth: 100,
+      treeAggregationType: uiGridGroupingConstants.aggregation.SUM
+  }, {
+      name: 'totals["Letter Sent"]["count"] || 0',
+      displayName: 'Letters',
+      minWidth: 100,
+      treeAggregationType: uiGridGroupingConstants.aggregation.SUM
+  }, {
+      name: 'totals["Home Visit"]["count"] || 0',
+      displayName: 'Visits',
+      minWidth: 100,
+      treeAggregationType: uiGridGroupingConstants.aggregation.SUM
+  }, {
+      name: 'totals["SST Referral"]["count"] || 0',
+      displayName: 'SST',
+      minWidth: 100,
+      treeAggregationType: uiGridGroupingConstants.aggregation.SUM
+  }, {
+      name: 'totals["Court Referral"]["count"] || 0',
+      displayName: 'Court',
+      minWidth: 100,
+      treeAggregationType: uiGridGroupingConstants.aggregation.SUM
+  }];
+
+  $scope.outreachesGridOptions.onRegisterApi = function(gridApi) {
+    $scope.outreachesGridApi = gridApi;
+    $scope.outreachesGridOptions.data = 
+      Student.outreachSummary();
+
+    $scope.outreachesGridOptions.data.$promise.then(function(data) {
+      // Convert counts array to object, generate total intervention property
+      _.forEach(data, function(student) {
+        student.totals = _.keyBy(student.counts, 'type');
```

Comment:
Nice. This looks good. One last thing is you could do something like:

```
        student.totals = _(student.counts)
          .keyBy('type').mapValues('count').value();
        student.totals.all = _.sumBy(student.counts, 'count');
```

And then you can drop all the `["count"]` in the `colDef.name`s.